config/config.go

```diff
@@ -132,6 +134,8 @@ type PersistConfig struct {
 	NodeStatusOffload bool `json:"nodeStatusOffLoad,omitempty"`
 	// Archive workflows to persistence.
 	Archive bool `json:"archive,omitempty"`
+	// ArchivelabelSelector holds LabelSelector to determine workflow persistence.
+	ArchiveLabelSelector *metav1.LabelSelector `json:"archiveLabelSelector,omitempty"`
```

Comment:
Shouldn't this be a `string` field? `metav1.LabelSelector` is a complicated struct:

```go
type LabelSelector struct {
	// matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
	// map is equivalent to an element of matchExpressions, whose key field is "key", the
	// operator is "In", and the values array contains only "value". The requirements are ANDed.
	// +optional
	MatchLabels map[string]string `json:"matchLabels,omitempty" protobuf:"bytes,1,rep,name=matchLabels"`
	// matchExpressions is a list of label selector requirements. The requirements are ANDed.
	// +optional
	MatchExpressions []LabelSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,2,rep,name=matchExpressions"`
}
```

We shouldn't expect our users to fill this complicated field out when we can simply have them use a string and parse it?

Reply:
This is a well known structure in Kubernetes.  I don't think it is not a complex structure for the operator. it used in Deployment and Service.
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#:~:text=A%20label%20selector%20can%20be,logical%20AND%20(%20%26%26%20)%20operator.&text=Caution%3A%20For%20both%20equality%2Dbased,logical%20OR%20(%20%7C%7C%20)%20operator.
```
selector:
  matchLabels:
    component: redis
  matchExpressions:
    - {key: tier, operator: In, values: [cache]}
    - {key: environment, operator: NotIn, values: [dev]}
```

Reply:
Got it, thanks for clarifying